### PR TITLE
build raylib from sources

### PR DESCRIPTION
**Pull request by anggape**
this will solve #118 it also has android build still work in progress https://github.com/anggape/Raylib-cs/tree/android, currently i only tested on windows-x64 and linux-x64, i dont have any macos laying around

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,150 +1,189 @@
...
+        run: |
+          curl -Lso raylib.zip https://github.com/raysan5/raylib/archive/refs/tags/${version}.zip
+          unzip -qq raylib.zip
+          cmake -S raylib-${version} \
+            -B build \
+            -D CMAKE_BUILD_TYPE=Release \
+            -D BUILD_SHARED_LIBS=ON \
+            -D BUILD_EXAMPLES=OFF
+          cmake --build build --config Release
+          file build/raylib/libraylib.so # TODO: remove me
```
Comment (9ParsonsB) [inline .github/workflows/build.yml:51]: Is this still required?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,150 +1,189 @@
...
+          curl -Lso raylib.zip https://github.com/raysan5/raylib/archive/refs/tags/${version}.zip
+          unzip -qq raylib.zip
+          cmake -S raylib-${version} \
+            -B build \
+            -D CMAKE_BUILD_TYPE=Release \
+            -D CMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} \
+            -D BUILD_SHARED_LIBS=ON \
+            -D BUILD_EXAMPLES=OFF
+          cmake --build build --config Release
+          file build/raylib/libraylib.dylib # TODO: remove me
```
Comment (9ParsonsB) [inline .github/workflows/build.yml:85]: Is this still required?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -1,150 +1,189 @@
 name: Build
 on:
   push:
-    branches: master
   pull_request:
-    branches: '*'
-    paths-ignore:
-      - 'README.md'
-      - '.github/ISSUE_TEMPLATE/**'
```
Comment (9ParsonsB) [inline .github/workflows/build.yml:9]: Is there any way we can keep this?

**Review by 9ParsonsB — APPROVED** [review]

**Review by 9ParsonsB — APPROVED** [review]

**Review by ChrisDill — APPROVED** [review]
Looks good! Feel free to squash merge when ready.

**Pull request merged**